src/main/java/racing/RacingGame.java

```diff
@@ -0,0 +1,54 @@
+package racing;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class RacingGame {
+
+    private final List<Car> cars;
+    private final MovingStrategy movingStrategy;
+
+    public RacingGame(List<Car> cars, MovingStrategy movingStrategy) {
+        this.cars = cars;
+        this.movingStrategy = movingStrategy;
+    }
+
+    public RacingGame(String carNames) {
+        this(carNames.split(","));
+    }
+
+    public RacingGame(String[] carNames) {
+        this(Arrays.stream(carNames)
+                .map(Car::new)
+                .collect(Collectors.toList()));
+    }
+    public RacingGame(List<Car> cars) {
+        this(cars, new RandomStrategy());
+    }
```

Comment:
생성자 오버로딩(특히나 이렇게 체이닝 형식으로 되있는경우)은 아래와 같이 생성자가 호출되는 순서대로둔다면
코드를 파악하기가 조금 더 쉬울 것 같아요. 

사소해보이지만 생성자가 지금처럼 위치하면 코드를 이리저리 왔다갔다하면서 파악해야해서, 코드를 파악하기가 조금 어려워지는 것 같아요. 🥲

```suggestion
    public RacingGame(List<Car> cars, MovingStrategy movingStrategy) {
        this.cars = cars;
        this.movingStrategy = movingStrategy;
    }

    public RacingGame(List<Car> cars) {
        this(cars, new RandomStrategy());
    }

    public RacingGame(String[] carNames) {
        this(Arrays.stream(carNames)
            .map(Car::new)
            .collect(Collectors.toList()));
    }
    public RacingGame(String carNames) {
        this(carNames.split(","));
    }
```